src/main/java/DAO/AbstractDao.java

```diff
@@ -0,0 +1,92 @@
+package DAO;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.exception.ConstraintViolationException;
+
+import javax.persistence.RollbackException;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class AbstractDao<T> implements Dao<T> {
+
+    private List<T> list = new LinkedList<>();
+
+    protected final Class<T> clazz;
+
+    protected SessionFactory sessionFactory;
+
+    protected Session getCurrentSession() {
+        return sessionFactory.getCurrentSession();
+    }
+
+    public AbstractDao(SessionFactory sessionFactory, Class<T> clazz){
+        this.clazz = clazz;
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Optional<T> findById(Integer id) throws SQLException {
+        Session session = sessionFactory.openSession();
+        return Optional.ofNullable(session.get(clazz,id));
+    }
+
+    public List<T> findAll() {
+        Session session = sessionFactory.openSession();
+        List allToList = session.createQuery("FROM " + clazz.getSimpleName()).list();
+        return allToList;
+    }
+
+    public Optional<T> save(T entity) throws ConstraintViolationException, RollbackException {
+        try (Session session = sessionFactory.openSession()) {
+            Transaction t1 = session.beginTransaction();
+            session.save(entity);
+            t1.commit();
+            return Optional.of(entity);
+        } catch (RollbackException e) {
+            Throwable t = e.getCause();
+            e.printStackTrace();
+            while ((t != null) && !(t instanceof ConstraintViolationException)){
+                t = t.getCause();
+                System.out.println("Dublicate");
+            }
+            if(t instanceof ConstraintViolationException){
+                Throwable finalT = t;
+                System.out.println("Dublicate" +t);
+            }
+        }
+        finally {
+            return Optional.of(entity);
```

Comment:
your method always returns an Optional.of(entity) doesn't matter what the result is.